### PR TITLE
Added the "MaxProtocol" method to set the maximum version for SMB connection.

### PR DESCRIPTION
**Pull request by mdoliveira**
Microsoft has advised its clients that version 1 of SMB is disabled on servers, but with this method it is possible to set the version. Example: 2 or 3.

### CODE REVIEW

**Comment by mdoliveira** [comment]
I found a request for this in the problems...
https://github.com/icewind1991/SMB/issues/55

**Comment by icewind1991** [comment]
I'm hesitant to add this until php-smbclient can support the same thing

**Comment by mmattel** [comment]
@icewind1991 now it can, see https://github.com/icewind1991/SMB/issues/99 (New eduardok smbclient option: smbclient_client_protocols)

File: src/Server.php [hunk 1]
```diff
@@ -11,155 +12,177 @@
...
+     * @return integer
+     */
+    public function getMaxProtocol() {
+        return $this->maxProtocol;
+    }
+
+    /**
+     * @param integer $maxProtocol
+     */
+    public function setMaxProtocol($maxProtocol) {
```
Comment (mmattel) [inline src/Server.php:181]: This function would need an adoption because you can set NT1 and SMB1/2/3

**Comment by icewind1991** [comment]
I've implemented this differently in 3.3.0, thanks for the work anyway!

**Pull request closed without merging**